### PR TITLE
Update to be possible to work with new android build plugin

### PR DESCRIPTION
**Pull request by jfcamel**
## Summary

Updated gradle to 4.1

### CODE REVIEW

File: src/main/groovy/de/lemona/gradle/plugins/AndroidPlugin.groovy [hunk 1]
```diff
@@ -86,11 +86,11 @@ class AndroidPlugin implements Plugin<Project> {
         }
 
         // Sources to JavaDOC
-        source variant.javaCompile.source
+        source variant.javaCompiler.source
```
Comment (jfcamel) [inline src/main/groovy/de/lemona/gradle/plugins/AndroidPlugin.groovy:87]: BaseVariant.javaCompile was deprecated.
https://android.googlesource.com/platform/tools/base/+/master/build-system/gradle-core/src/main/groovy/com/android/build/gradle/api/BaseVariant.java#177

new one.
https://android.googlesource.com/platform/tools/base/+/master/build-system/gradle-core/src/main/groovy/com/android/build/gradle/api/BaseVariant.java#184

File: src/main/groovy/de/lemona/gradle/plugins/AndroidPlugin.groovy [hunk 2]
```diff
@@ -86,11 +86,11 @@ class AndroidPlugin implements Plugin<Project> {
...
 
         // Start with the "default" Android SDK classpath and add our variant classpath
         classpath = files("${android.sdkDirectory}/platforms/${android.compileSdkVersion}/android.jar")
-        classpath += files(variant.javaCompile.classpath.files)
+        classpath += files(variant.javaCompiler.classpath)
```
Comment (jfcamel) [inline src/main/groovy/de/lemona/gradle/plugins/AndroidPlugin.groovy:91]: deprecated

File: src/main/groovy/de/lemona/gradle/plugins/PublishPlugin.groovy [hunk 3]
```diff
@@ -163,12 +162,27 @@ class PublishPlugin implements Plugin<Project> {
 
             // Create our APK publication
             publishing.publications.create(_variant, MavenPublication) {
-              artifact _packageTask.outputFile
```
Comment (jfcamel) [inline src/main/groovy/de/lemona/gradle/plugins/PublishPlugin.groovy:166]: outputFile was deprecated

File: src/main/groovy/de/lemona/gradle/plugins/PublishPlugin.groovy [hunk 4]
```diff
@@ -163,12 +162,27 @@ class PublishPlugin implements Plugin<Project> {
...
-              pom {
-                packaging = 'apk'
-                withXml {
-                  asNode().appendNode('properties')
-                    .appendNode('versionCode', _versionCode)
+              _packageTask.outputs.files.each {
```
Comment (jfcamel) [inline src/main/groovy/de/lemona/gradle/plugins/PublishPlugin.groovy:165]: outputs is new one.

File: src/main/groovy/de/lemona/gradle/plugins/PublishPlugin.groovy [hunk 5]
```diff
@@ -163,12 +162,27 @@ class PublishPlugin implements Plugin<Project> {
...
+                logger.debug("debugInfo: output files {}", it.toString())
+                def files = it.listFiles(new FileFilter() {
```
Comment (jfcamel) [inline src/main/groovy/de/lemona/gradle/plugins/PublishPlugin.groovy:167]: not smart...

Comment (tkcmrc) [reply src/main/groovy/de/lemona/gradle/plugins/PublishPlugin.groovy:167]: ^^;

**Review by aostanin — APPROVED** [review]

**Review by tkcmrc — APPROVED** [review]

**Review by vedran55 — APPROVED** [review]

**Pull request merged**